cloudinit/sources/DataSourceOVF.py

```diff
@@ -259,21 +269,43 @@ def transport_iso9660(require_iso=True):
 
 
 def transport_vmware_guestinfo():
-    rpctool = "vmware-rpctool"
-    not_found = None
-    if not subp.which(rpctool):
-        return not_found
-    cmd = [rpctool, "info-get guestinfo.ovfEnv"]
+    is_vmtoolsd = False
```

Comment:
It's a little hard for me to follow the logic here, and it feels a little more strongly coupled than needed.

What about something like this:
```diff
diff --git a/cloudinit/sources/DataSourceOVF.py b/cloudinit/sources/DataSourceOVF.py
index 7b9bf65af..53a544458 100644
--- a/cloudinit/sources/DataSourceOVF.py
+++ b/cloudinit/sources/DataSourceOVF.py
@@ -269,29 +269,22 @@ def transport_iso9660(require_iso=True):
 
 
 def transport_vmware_guestinfo():
-    is_vmtoolsd = False
-    rpctool = subp.which("vmware-rpctool")
-
-    if not rpctool:
-        rpctool = subp.which("vmtoolsd")
-        is_vmtoolsd = True
-
-    if not rpctool:
-        return None
-
-    try:
-        return transport_vmware_guestinfo_rpctool(rpctool, is_vmtoolsd)
-    except subp.ProcessExecutionError:
-        rpctool = subp.which("vmtoolsd")
-        is_vmtoolsd = True
-        return transport_vmware_guestinfo_rpctool(rpctool, is_vmtoolsd)
+    rpctool_path = subp.which("vmware-rpctool")
+    if rpctool_path:
+        result = transport_vmware_guestinfo_rpctool(rpctool_path, "")
+        if result:
+            return result
+    vmtoolsd_path = subp.which("vmtoolsd")
+    if vmtoolsd_path:
+        return transport_vmware_guestinfo_rpctool(vmtoolsd_path, "--cmd")
+    return None
 
 
-def transport_vmware_guestinfo_rpctool(rpctool, is_vmtoolsd=False):
+def transport_vmware_guestinfo_rpctool(rpctool, cmd_arg=""):
     args = [rpctool]
-    if is_vmtoolsd:
-        args.append("--cmd")
-    args.append("info-get guestinfo.ovfEnv")
+    if cmd_arg:
+        args.append(cmd_arg)
+    args.extend(["info-get", "guestinfo.ovfEnv"])
 
     try:
         out, _err = subp.subp(args)
@@ -302,8 +295,6 @@ def transport_vmware_guestinfo_rpctool(rpctool, is_vmtoolsd=False):
         if e.exit_code != 1:
             LOG.warning("%s exited with code %d", rpctool, e.exit_code)
             LOG.debug(e)
-        if not is_vmtoolsd:
-            raise e
 
     return None
 
```

Reply:
Yeah, and if you feel strongly, I can do something like that. I wanted to minimize the refactor the DS I don't own. I know it has the same issue, hence why I took it on.

Reply:
Please note you are incorrectly assuming that `info-get` and the next arg are separate. They aren't. See the examples in the command lines in the description.

Reply:
> Yeah, and if you feel strongly, I can do something like that

If you agree I think it's preferable, but if you have qualms we don't need to go that way.

Reply:
or we combine a cmd directly for either rpctool, and give it to transport_vmware_guestinfo_rpctool(cmd) to execute it, sorry for bad indentation if any.

def transport_vmware_guestinfo():
    rpctool = subp.which("vmware-rpctool")
    if rpctool:
        cmd = [rpctool, "info-get guestinfo.ovfEnv"]
        result = transport_vmware_guestinfo_rpctool(cmd)
        if result:
            return result
    rpctools = subp.which("vmtoolsd")
    if rpctools:
        cmd = [rpctool, "--cmd", "info-get guestinfo.ovfEnv"]
        result = transport_vmware_guestinfo_rpctool(cmd)
        if result:
            return result
    return None

Reply:
I ended up resolving it a little differently than both of the above approaches. Interested in your thoughts.